java-manta-client/src/main/java/com/joyent/manta/http/StandardHttpHelper.java

```diff
@@ -351,13 +351,11 @@ protected static void validateChecksum(final DigestedEntity entity,
                                            final HttpRequest request,
                                            final HttpResponse response)
             throws MantaChecksumFailedException {
-        if (entity == null) {
-            return;
-        }
+        Validate.notNull(entity, "Request body required");
 
         if (serverMd5 == null || serverMd5.length == 0) {
-            LOGGER.warn("No cryptographic check performed by the server");
-            return;
+            final String msg = "Server calculated MD5 is missing";
```

Comment:
Correct. This method is only called on standard PUT requests while that ticket is about MPU commit.